### PR TITLE
build!: Update to net8.0 to fix canary

### PR DESCRIPTION
**Pull request by agneszitte**
## PR Type
- Build or CI-related changes

## Description

- Update to net8.0 to fix canary.
- **BREAKING CHANGE: Removing support for net7.**

### CODE REVIEW

File: src/library/Uno.Toolkit.WinUI.Markup/Uno.Toolkit.WinUI.Markup.csproj [hunk 1]
```diff
@@ -6,14 +6,14 @@
...
-		<TargetFrameworks Condition="'$(TargetFrameworkOverride)'=='' and '$(DisableNet7MobileTargets)'==''">$(TargetFrameworks);net7.0-ios;net7.0-macos;net7.0-android;net7.0-maccatalyst</TargetFrameworks>
-		<TargetFrameworks Condition="'$(TargetFrameworkOverride)'=='' and '$(OS)'=='Windows_NT'">$(TargetFrameworks);net7.0-windows10.0.19041</TargetFrameworks>
+		<TargetFrameworks Condition="'$(TargetFrameworkOverride)'==''">$(TargetFrameworks);net8.0;net8.0-ios;net8.0-macos;net8.0-android;net8.0-maccatalyst</TargetFrameworks>
+		<TargetFrameworks Condition="'$(TargetFrameworkOverride)'=='' and '$(OS)'=='Windows_NT'">$(TargetFrameworks);net8.0-windows10.0.19041</TargetFrameworks>
 		<AllowedOutputExtensionsInPackageBuildOutputFolder>.pdb</AllowedOutputExtensionsInPackageBuildOutputFolder>
 		<!-- Ensures the .xr.xml files are generated in a proper layout folder -->
 		<GenerateLibraryLayout>true</GenerateLibraryLayout>
 		<DefineConstants>WINUI</DefineConstants>
 		<UseWinUI>true</UseWinUI>
+		<RuntimeIdentifiers>win-x86;win-x64;win-arm64</RuntimeIdentifiers>
```
Comment (Youssef1313) [inline src/library/Uno.Toolkit.WinUI.Markup/Uno.Toolkit.WinUI.Markup.csproj:16]: This should be for the Windows build only.

**Comment by agneszitte** [comment]
@jeromelaban, @Youssef1313 on my last push I tried to keep net7 just for iOS UI Test just for info. But we can re-adjust it differently depending on what solution we will come up with for UI Tests here

**Comment by Youssef1313** [comment]
> On my last push I tried to keep net7 just for iOS UI Test just for info

It's unlikely this will work. Even if it passes CI in this PR, the canary build will fail as the Uno package no longer supports net7.0

**Comment by Youssef1313** [comment]
Working on this in #1154

**Comment by agneszitte** [comment]
Closing this PR in favor of https://github.com/unoplatform/uno.toolkit.ui/pull/1154

**Pull request closed without merging**